documentation/src/main/asciidoc/performance_guide/performance_guide.asciidoc

```diff
@@ -0,0 +1,197 @@
+= Infinispan {infinispanversion} Performance Guide
+The Infinispan community
+:toc2:
+:icons: font
+:toclevels: 3
+:numbered:
+:stem:
+
+This guide will give you information and tweaks about tuning Infinispan performance (both server and library mode).
+
+== Capacity planning
+
+Infinispan stores data in memory. In case of Local or Replicated mode, all data needs to fit in there,
+so calculating the amount of required memory is trivial.
+
```

Comment:
It'd be interesting to put some guidelines in order to estimate or calculate the amount of memory. Need to take into account the (serialized or not) key size, (serialized or not) value size, the Infinispan entry overhead (200 bytes?), and the JVM overhead (object header, alignment, etc). This can be done with estimation using tools like http://openjdk.java.net/projects/code-tools/jol/ or actually measuring the heap with a subset of data.